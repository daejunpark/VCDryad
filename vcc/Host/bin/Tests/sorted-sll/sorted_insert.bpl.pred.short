ensures b0000 ==> (F#srtl($s,$phys_ptr_cast(P#x,^s_node)));
ensures b0001 ==> (F#srtl($s,$phys_ptr_cast($result,^s_node)));
ensures b0002 ==> ((F#srtl_reach($s,$phys_ptr_cast(P#x,^s_node)) == F#srtl_reach($s,$phys_ptr_cast($result,^s_node))));
ensures b0003 ==> ((F#srtl_reach($s,$phys_ptr_cast($result,^s_node)) == F#srtl_reach($s,$phys_ptr_cast(P#x,^s_node))));
ensures b0004 ==> ((F#srtl_reach($s,$phys_ptr_cast(P#x,^s_node)) == F#srtl_reach(old($s),$phys_ptr_cast(P#x,^s_node))));
ensures b0005 ==> ((F#srtl_reach($s,$phys_ptr_cast($result,^s_node)) == F#srtl_reach(old($s),$phys_ptr_cast($result,^s_node))));
ensures b0006 ==> ($non_null($phys_ptr_cast(P#x,^s_node)));
ensures b0007 ==> ($non_null($phys_ptr_cast($result,^s_node)));
ensures b0008 ==> ($is_null($phys_ptr_cast(P#x,^s_node)));
ensures b0009 ==> ($is_null($phys_ptr_cast($result,^s_node)));
ensures b0010 ==> (($phys_ptr_cast(P#x,^s_node) == $phys_ptr_cast($result,^s_node)));
ensures b0011 ==> (($phys_ptr_cast($result,^s_node) == $phys_ptr_cast(P#x,^s_node)));
ensures b0012 ==> (($non_null($phys_ptr_cast(P#x,^s_node)) ==> $non_null($rd_phys_ptr($s,s_node.next,$phys_ptr_cast(P#x,^s_node),^s_node))));
ensures b0013 ==> (($non_null($phys_ptr_cast($result,^s_node)) ==> $non_null($rd_phys_ptr($s,s_node.next,$phys_ptr_cast($result,^s_node),^s_node))));
ensures b0014 ==> (($non_null($phys_ptr_cast(P#x,^s_node)) ==> $is_null($rd_phys_ptr($s,s_node.next,$phys_ptr_cast(P#x,^s_node),^s_node))));
ensures b0015 ==> (($non_null($phys_ptr_cast($result,^s_node)) ==> $is_null($rd_phys_ptr($s,s_node.next,$phys_ptr_cast($result,^s_node),^s_node))));
ensures b0016 ==> (($non_null($phys_ptr_cast(P#x,^s_node)) ==> ($rd_phys_ptr($s,s_node.next,$phys_ptr_cast(P#x,^s_node),^s_node) == $phys_ptr_cast($result,^s_node))));
ensures b0017 ==> (($non_null($phys_ptr_cast($result,^s_node)) ==> ($rd_phys_ptr($s,s_node.next,$phys_ptr_cast($result,^s_node),^s_node) == $phys_ptr_cast(P#x,^s_node))));
ensures b0018 ==> ((F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)) == F#sll_keys($s,$phys_ptr_cast($result,^s_node))));
ensures b0019 ==> ((F#sll_keys($s,$phys_ptr_cast($result,^s_node)) == F#sll_keys($s,$phys_ptr_cast(P#x,^s_node))));
ensures b0020 ==> ((F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)) == F#sll_keys(old($s),$phys_ptr_cast($result,^s_node))));
ensures b0021 ==> ((F#sll_keys($s,$phys_ptr_cast($result,^s_node)) == F#sll_keys(old($s),$phys_ptr_cast(P#x,^s_node))));
ensures b0022 ==> ((F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)) == $intset_union(F#sll_keys(old($s),$phys_ptr_cast($result,^s_node)),$intset_singleton(P#k))));
ensures b0023 ==> ((F#sll_keys($s,$phys_ptr_cast($result,^s_node)) == $intset_union(F#sll_keys(old($s),$phys_ptr_cast(P#x,^s_node)),$intset_singleton(P#k))));
ensures b0024 ==> ((F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)) == F#sll_keys($s,$phys_ptr_cast($result,^s_node))));
ensures b0025 ==> ((F#sll_keys($s,$phys_ptr_cast($result,^s_node)) == F#sll_keys($s,$phys_ptr_cast(P#x,^s_node))));
ensures b0026 ==> ((F#sll_keys(old($s),$phys_ptr_cast(P#x,^s_node)) == $intset_union(F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)),F#sll_keys($s,$phys_ptr_cast($result,^s_node)))));
ensures b0027 ==> ((F#sll_keys(old($s),$phys_ptr_cast($result,^s_node)) == $intset_union(F#sll_keys($s,$phys_ptr_cast($result,^s_node)),F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)))));
ensures b0028 ==> ((F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)) == $intset_union(F#sll_keys(old($s),$phys_ptr_cast(P#x,^s_node)),F#sll_keys($s,$phys_ptr_cast($result,^s_node)))));
ensures b0029 ==> ((F#sll_keys($s,$phys_ptr_cast($result,^s_node)) == $intset_union(F#sll_keys(old($s),$phys_ptr_cast($result,^s_node)),F#sll_keys($s,$phys_ptr_cast(P#x,^s_node)))));
ensures b0030 ==> ((P#k < 2147483647));
ensures b0031 ==> ((P#k < 2147483647));
ensures b0032 ==> ((P#k < 4294967295));
ensures b0033 ==> ((P#k < 4294967295));
ensures b0034 ==> ((P#k >= 0));
ensures b0035 ==> ((P#k >= 0));
ensures b0036 ==> (($rd_inv($s,s_node.key,$phys_ptr_cast(P#x,^s_node)) < P#k));
ensures b0037 ==> (($rd_inv($s,s_node.key,$phys_ptr_cast($result,^s_node)) < P#k));
ensures b0038 ==> (($rd_inv($s,s_node.key,$phys_ptr_cast(P#x,^s_node)) <= $rd_inv($s,s_node.key,$phys_ptr_cast($result,^s_node))));
ensures b0039 ==> (($rd_inv($s,s_node.key,$phys_ptr_cast($result,^s_node)) <= $rd_inv($s,s_node.key,$phys_ptr_cast(P#x,^s_node))));
ensures b0040 ==> (($rd_inv($s,s_node.key,$phys_ptr_cast(P#x,^s_node)) == $rd_inv($s,s_node.key,$phys_ptr_cast($result,^s_node))));
ensures b0041 ==> (($rd_inv($s,s_node.key,$phys_ptr_cast($result,^s_node)) == $rd_inv($s,s_node.key,$phys_ptr_cast(P#x,^s_node))));
INV
const {:existential true} b0000 : bool;
const {:existential true} b0001 : bool;
const {:existential true} b0002 : bool;
const {:existential true} b0003 : bool;
const {:existential true} b0004 : bool;
const {:existential true} b0005 : bool;
const {:existential true} b0006 : bool;
const {:existential true} b0007 : bool;
const {:existential true} b0008 : bool;
const {:existential true} b0009 : bool;
const {:existential true} b0010 : bool;
const {:existential true} b0011 : bool;
const {:existential true} b0012 : bool;
const {:existential true} b0013 : bool;
const {:existential true} b0014 : bool;
const {:existential true} b0015 : bool;
const {:existential true} b0016 : bool;
const {:existential true} b0017 : bool;
const {:existential true} b0018 : bool;
const {:existential true} b0019 : bool;
const {:existential true} b0020 : bool;
const {:existential true} b0021 : bool;
const {:existential true} b0022 : bool;
const {:existential true} b0023 : bool;
const {:existential true} b0024 : bool;
const {:existential true} b0025 : bool;
const {:existential true} b0026 : bool;
const {:existential true} b0027 : bool;
const {:existential true} b0028 : bool;
const {:existential true} b0029 : bool;
const {:existential true} b0030 : bool;
const {:existential true} b0031 : bool;
const {:existential true} b0032 : bool;
const {:existential true} b0033 : bool;
const {:existential true} b0034 : bool;
const {:existential true} b0035 : bool;
const {:existential true} b0036 : bool;
const {:existential true} b0037 : bool;
const {:existential true} b0038 : bool;
const {:existential true} b0039 : bool;
const {:existential true} b0040 : bool;
const {:existential true} b0041 : bool;
