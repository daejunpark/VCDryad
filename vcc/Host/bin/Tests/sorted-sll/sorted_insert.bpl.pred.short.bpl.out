
Z:\InvariantSynthesis\VCDryad\vcc\Host\bin\Tests>C:\boogie\Boogie\Boogie.exe /typeEncoding:m /prover:SMTLib /useArrayTheory /trace /noinfer /contractInfer Z:\InvariantSynthesis\repo\vcdryad_bpl\Vcc3Prelude.new.bpl sorted-sll\sorted_insert.bpl.pred.bpl 
[TRACE] Using prover: C:\boogie\Boogie\z3.exe
Boogie program verifier version 2.2.30705.1126, Copyright (c) 2003-2013, Microsoft.
Parsing Z:\InvariantSynthesis\repo\vcdryad_bpl\Vcc3Prelude.new.bpl
Parsing sorted-sll\sorted_insert.bpl.pred.bpl
Coalescing blocks...
Collecting existential constants...
Building call graph...
Number of implementations = 1
Beginning VC generation for Houdini...
Generating VC for sorted_insert
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 b0002 b0003 b0004 b0005 b0006 b0007 b0008 b0009 b0010 b0011 b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 1.1231064
Removing b0002
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) b0003 b0004 b0005 b0006 b0007 b0008 b0009 b0010 b0011 b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2183818
Removing b0003
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 b0005 b0006 b0007 b0008 b0009 b0010 b0011 b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 3.6344971
Removing b0005
Removing b0006
Removing b0008
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) b0009 b0010 b0011 b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2339805
Removing b0009
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) b0010 b0011 b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2339805
Removing b0010
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) b0011 b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2339805
Removing b0011
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) b0012 b0013 b0014 b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 7.9397383
Removing b0013
Removing b0014
Removing b0012
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) b0015 b0016 b0017 b0018 b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 4.8511957
Removing b0018
Removing b0015
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) b0016 b0017 (! b0018) b0019 b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 4.1804516
Removing b0019
Removing b0016
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) b0020 b0021 b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 2.7765686
Removing b0020
Removing b0021
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) b0022 b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2651779
Removing b0022
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 b0024 b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2495792
Removing b0024
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) b0025 b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2339805
Removing b0025
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) b0026 b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.2339805
Removing b0026
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) b0004 (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) (! b0026) b0027 b0028 b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 1.2010999
Removing b0027
Removing b0028
Removing b0004
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) (! b0004) (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) (! b0026) (! b0027) (! b0028) b0029 b0030 b0031 b0032 b0033 b0034 b0035 b0036 b0037 b0038 b0039 b0040 b0041)
Time taken = 0.8579285
Removing b0029
Removing b0034
Removing b0036
Removing b0030
Removing b0037
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) (! b0004) (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) b0017 (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) (! b0026) (! b0027) (! b0028) (! b0029) (! b0030) b0031 b0032 b0033 (! b0034) b0035 (! b0036) (! b0037) b0038 b0039 b0040 b0041)
Time taken = 1.091909
Removing b0035
Removing b0038
Removing b0031
Removing b0039
Removing b0017
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) (! b0004) (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) (! b0017) (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) (! b0026) (! b0027) (! b0028) (! b0029) (! b0030) (! b0031) b0032 b0033 (! b0034) (! b0035) (! b0036) (! b0037) (! b0038) (! b0039) b0040 b0041)
Time taken = 0.4367636
Removing b0040
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) (! b0004) (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) (! b0017) (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) (! b0026) (! b0027) (! b0028) (! b0029) (! b0030) (! b0031) b0032 b0033 (! b0034) (! b0035) (! b0036) (! b0037) (! b0038) (! b0039) (! b0040) b0041)
Time taken = 0.3899675
Removing b0041
Verifying sorted_insert
Houdini assignment axiom: (And true b0000 b0001 (! b0002) (! b0003) (! b0004) (! b0005) (! b0006) b0007 (! b0008) (! b0009) (! b0010) (! b0011) (! b0012) (! b0013) (! b0014) (! b0015) (! b0016) (! b0017) (! b0018) (! b0019) (! b0020) (! b0021) (! b0022) b0023 (! b0024) (! b0025) (! b0026) (! b0027) (! b0028) (! b0029) (! b0030) (! b0031) b0032 b0033 (! b0034) (! b0035) (! b0036) (! b0037) (! b0038) (! b0039) (! b0040) (! b0041))
Time taken = 0.2027831
Number of true assignments = 6
Number of false assignments = 36
Prover time = 30.59
Unsat core prover time = 0.00
Number of prover queries = 20
Number of unsat core prover queries = 0
Number of unsat core prunings = 0
verified

Boogie program verifier finished with 1 verified, 0 errors
